<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1098</width>
    <height>732</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Benjamins Display Tool</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabShape">
       <enum>QTabWidget::Triangular</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Full Frame</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="DispEditor" name="fullEditor" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QGroupBox" name="overlayBox">
            <property name="title">
             <string>Draw Overlay</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <property name="leftMargin">
              <number>6</number>
             </property>
             <property name="topMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>6</number>
             </property>
             <property name="bottomMargin">
              <number>6</number>
             </property>
             <property name="spacing">
              <number>3</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="toolTip">
                <string>Position to draw image</string>
               </property>
               <property name="text">
                <string>Position</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1" colspan="2">
              <widget class="QDoubleSpinBox" name="ovRotBox">
               <property name="suffix">
                <string> Deg</string>
               </property>
               <property name="minimum">
                <double>-360.000000000000000</double>
               </property>
               <property name="maximum">
                <double>360.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="3" column="2">
              <widget class="QSpinBox" name="ovImCYPosBox">
               <property name="prefix">
                <string>Y: </string>
               </property>
               <property name="minimum">
                <number>-1024</number>
               </property>
               <property name="maximum">
                <number>1024</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_4">
               <property name="toolTip">
                <string>This pixel on the image will correspond to the position above</string>
               </property>
               <property name="text">
                <string>Img Center</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QSpinBox" name="ovXPosBox">
               <property name="prefix">
                <string>X: </string>
               </property>
               <property name="minimum">
                <number>-255</number>
               </property>
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_5">
               <property name="toolTip">
                <string>Rotation will be done around this pixel on the image</string>
               </property>
               <property name="text">
                <string>Rot Center</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="ovImCXPosBox">
               <property name="prefix">
                <string>X: </string>
               </property>
               <property name="minimum">
                <number>-1024</number>
               </property>
               <property name="maximum">
                <number>1024</number>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QSpinBox" name="ovCrHBox">
               <property name="prefix">
                <string>H: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
               <property name="value">
                <number>128</number>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QSpinBox" name="ovYPosBox">
               <property name="prefix">
                <string>Y: </string>
               </property>
               <property name="minimum">
                <number>-255</number>
               </property>
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="label_6">
               <property name="toolTip">
                <string>Image rotation in degrees</string>
               </property>
               <property name="text">
                <string>Rotation</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QSpinBox" name="ovRXPosBox">
               <property name="prefix">
                <string>X: </string>
               </property>
               <property name="minimum">
                <number>-1024</number>
               </property>
               <property name="maximum">
                <number>1024</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_2">
               <property name="toolTip">
                <string>Corner to start crop. The image will not be drawn outside of the crop area. Cropping the drawing greatly improves performance.</string>
               </property>
               <property name="text">
                <string>Crop Start</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_3">
               <property name="toolTip">
                <string>Size of cropped area</string>
               </property>
               <property name="text">
                <string>Crop Size</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QSpinBox" name="ovCrYPosBox">
               <property name="prefix">
                <string>Y: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="ovCrWBox">
               <property name="prefix">
                <string>W: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
               <property name="value">
                <number>128</number>
               </property>
              </widget>
             </item>
             <item row="7" column="1" colspan="2">
              <widget class="QSpinBox" name="ovTrBox">
               <property name="toolTip">
                <string>Color index on image to count as transparent.</string>
               </property>
               <property name="minimum">
                <number>-1</number>
               </property>
               <property name="maximum">
                <number>15</number>
               </property>
               <property name="value">
                <number>0</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="ovCrXPosBox">
               <property name="prefix">
                <string>X: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
              </widget>
             </item>
             <item row="4" column="2">
              <widget class="QSpinBox" name="ovRYPosBox">
               <property name="prefix">
                <string>Y: </string>
               </property>
               <property name="minimum">
                <number>-1024</number>
               </property>
               <property name="maximum">
                <number>1024</number>
               </property>
              </widget>
             </item>
             <item row="8" column="0" colspan="3">
              <widget class="QPushButton" name="ovSaveButton">
               <property name="toolTip">
                <string>Save overlay to layer 2 on image. Layer 2 can be enabled and disabled independently.</string>
               </property>
               <property name="text">
                <string>Save to Layer 2</string>
               </property>
              </widget>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Transparent</string>
               </property>
              </widget>
             </item>
             <item row="6" column="0">
              <widget class="QLabel" name="label_7">
               <property name="toolTip">
                <string>Image scale</string>
               </property>
               <property name="text">
                <string>Scale</string>
               </property>
              </widget>
             </item>
             <item row="6" column="1" colspan="2">
              <widget class="QDoubleSpinBox" name="ovScaleBox">
               <property name="decimals">
                <number>3</number>
               </property>
               <property name="maximum">
                <double>100.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="fontOverlayBox">
            <property name="title">
             <string>Draw Font</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <property name="leftMargin">
              <number>6</number>
             </property>
             <property name="topMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>6</number>
             </property>
             <property name="bottomMargin">
              <number>6</number>
             </property>
             <property name="spacing">
              <number>3</number>
             </property>
             <item row="1" column="0">
              <widget class="QCheckBox" name="fontBoldBox">
               <property name="text">
                <string>Bold</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0" colspan="2">
              <widget class="QFontComboBox" name="fontBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item row="11" column="0" colspan="2">
              <widget class="QPushButton" name="exportFontAAButton">
               <property name="text">
                <string>Export Antialiased C File</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QSpinBox" name="fontYPosBox">
               <property name="prefix">
                <string>Y: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
               <property name="value">
                <number>46</number>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QSpinBox" name="fontHBox">
               <property name="prefix">
                <string>H: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
               <property name="value">
                <number>16</number>
               </property>
              </widget>
             </item>
             <item row="7" column="0" colspan="2">
              <widget class="QLineEdit" name="fontSampleEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>ABC 00</string>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QSpinBox" name="fontWBox">
               <property name="prefix">
                <string>W: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QSpinBox" name="fontXPosBox">
               <property name="prefix">
                <string>X: </string>
               </property>
               <property name="maximum">
                <number>128</number>
               </property>
               <property name="value">
                <number>34</number>
               </property>
              </widget>
             </item>
             <item row="6" column="0" colspan="2">
              <widget class="QDoubleSpinBox" name="fontScaleBox">
               <property name="prefix">
                <string>Scale: </string>
               </property>
               <property name="decimals">
                <number>2</number>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.010000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="10" column="0" colspan="2">
              <widget class="QPushButton" name="exportFontButton">
               <property name="text">
                <string>Export C File</string>
               </property>
              </widget>
             </item>
             <item row="9" column="0">
              <widget class="QCheckBox" name="fontAABox">
               <property name="text">
                <string>Antialias</string>
               </property>
              </widget>
             </item>
             <item row="9" column="1">
              <widget class="QCheckBox" name="fontBorderBox">
               <property name="toolTip">
                <string>Draw border around characters to see their bounding box</string>
               </property>
               <property name="text">
                <string>Border</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Custom</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QSpinBox" name="wBox">
            <property name="prefix">
             <string>W: </string>
            </property>
            <property name="maximum">
             <number>1024</number>
            </property>
            <property name="value">
             <number>128</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="hBox">
            <property name="prefix">
             <string>H: </string>
            </property>
            <property name="maximum">
             <number>1024</number>
            </property>
            <property name="value">
             <number>64</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="updateSizeButton">
            <property name="text">
             <string>Update</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="DispEditor" name="customEditor" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1098</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionShow_instruction_sheet"/>
   </widget>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionExport_C_array">
   <property name="text">
    <string>Export C array</string>
   </property>
  </action>
  <action name="actionSave_PNG">
   <property name="text">
    <string>Save PNG</string>
   </property>
  </action>
  <action name="actionOpen_PNG">
   <property name="text">
    <string>Open PNG</string>
   </property>
  </action>
  <action name="actionShow_instruction_sheet">
   <property name="text">
    <string>Show instruction sheet</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>DispEditor</class>
   <extends>QWidget</extends>
   <header>dispeditor.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
